### PR TITLE
[monitor query] metrics batch GA 

### PR DESCRIPTION
**Pull request by gracewilcox**
Regenerated Monitor Query package from the newly released GA swaggers. The new swaggers contain some additional optional fields for metric queries. 

I also modified the docs, tests, and examples. The autorest file is now modified to generate properly with the new swaggers.

### CODE REVIEW

File: sdk/monitor/azquery/README.md [hunk 1]
```diff
@@ -145,6 +147,7 @@ comments.
 [default_cred_ref]: https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/azidentity#defaultazurecredential
 [example_logs_client]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-NewLogsClient
 [example_metrics_client]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-NewMetricsClient
+[example_metrics_batch_client]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery@v1.2.0-beta.1#example-NewMetricsBatchClient
```
Comment (chlowell) [inline sdk/monitor/azquery/README.md:150]: Ultimately you want to link to the example without specifying a version so clickers always land on the latest stable docs. But the example doesn't exist in the stable docs yet, so you have some options here:
1. remember to remove the version from this link after the 1.0 release
2. remove the version now. The link will take clickers to the top of the docs today and, after the release, to the example
3. use an aka link with fragment e.g. `https://aka.ms/azsdk/go/monitor/azquery#example-NewMetricsBatchClient`

I like option 2

File: sdk/monitor/azquery/autorest.md [hunk 2]
```diff
@@ -160,11 +170,30 @@ directive:
...
-    - https://github.com/Azure/azure-rest-api-specs/blob/21f5332f2dc7437d1446edf240e9a3d4c90c6431/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-01-01/metrics_API.json
-    - https://github.com/Azure/azure-rest-api-specs/blob/21f5332f2dc7437d1446edf240e9a3d4c90c6431/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
+    - https://github.com/Azure/azure-rest-api-specs/blob/0373f0edc4414fd402603fac51d0df93f1f70507/specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/metricDefinitions_API.json
+    - https://github.com/Azure/azure-rest-api-specs/blob/0373f0edc4414fd402603fac51d0df93f1f70507/specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/metrics_API.json
+    - https://github.com/Azure/azure-rest-api-specs/blob/0373f0edc4414fd402603fac51d0df93f1f70507/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
 
 directive:
+  # Remove subscription scope operations. Functionality not currenlty exposed
```
Comment (chlowell) [inline sdk/monitor/azquery/autorest.md:178]:
```suggestion
  # Remove subscription scope operations. Functionality not currently exposed
```

File: sdk/monitor/azquery/autorest.md [hunk 3]
```diff
@@ -195,24 +224,23 @@ directive:
...
+
 
   # change type of MetricsClientQueryResourceOptions.Aggregation from *string to []*AggregationType
   - from: options.go
     where: $
-    transform: return $.replace(/Aggregation \*string/g, "Aggregation []*AggregationType");
-  - from: 
-        - metrics_client.go
-        - metricsbatch_client.go
+    transform: return $.replace(/MetricsClientQueryResourceOptions struct {\n\t(?:\/\/.*\s)+\tAggregation \*string/g, "MetricsClientQueryResourceOptions struct {\n	// The list of aggregation types to retrieve\n  Aggregation []*AggregationType");
```
Comment (chlowell) [inline sdk/monitor/azquery/autorest.md:235]: Why do this for `MetricsClientQueryResourceOptions` but not `MetricsBatchClientQueryBatchOptions`?

File: sdk/monitor/azquery/autorest.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/autorest.md]: fixed in [b9b76fa](https://github.com/Azure/azure-sdk-for-go/pull/22243/commits/b9b76fa15f241675fd7abf6fb0a063a5faa8b1f4)

File: sdk/monitor/azquery/README.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/README.md]: I think you're right. I like option 2 the best. Changed in [b9b76fa](https://github.com/Azure/azure-sdk-for-go/pull/22243/commits/b9b76fa15f241675fd7abf6fb0a063a5faa8b1f4)

File: sdk/monitor/azquery/autorest.md (hunk 3 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/autorest.md:235]: Changing the type from its original string to []*AggregationType is added convenience to the user, but there's potential for issues in the future. I've gotten more conservative in my design decisions, so I would prefer to stick to the swagger.

MetricsClientQueryResourceOptions is already GA so can't change it, but I can modify it for MetricsBatchClientQueryBatchOptions.

Comment (chlowell) [reply sdk/monitor/azquery/autorest.md:235]: That makes sense but gives us two different types for the same kind of value. Is that inconsistency okay? (I don't have a strong opinion)

File: sdk/monitor/azquery/CHANGELOG.md [hunk 4]
```diff
@@ -3,8 +3,12 @@
 ## 1.2.0-beta.2 (Unreleased)
```
Comment (chlowell) [inline sdk/monitor/azquery/CHANGELOG.md:3]: Is the plan to ship v1.0 this month?

Comment (gracewilcox) [reply sdk/monitor/azquery/CHANGELOG.md]: Planning to GA end of January or February. Was going to update the version when prepping for release, but can do it.

**Pull request closed without merging**